src/server/snapshot.cc

```diff
@@ -332,8 +349,19 @@ bool SliceSnapshot::PushSerializedToChannel(bool force) {
 void SliceSnapshot::OnDbChange(DbIndex db_index, const DbSlice::ChangeReq& req) {
   FiberAtomicGuard fg;
   PrimeTable* table = db_slice_->GetTables(db_index).first;
+  const PrimeTable::bucket_iterator* bit = req.update();
+
+  if (bit && (bit->GetVersion() >= snapshot_version_)) {
+    return;
+  }
+
+  // If a bucket is being serialized we need to block
+  // maybe worth adding it on an event queue?
```

Comment:
We can notify and wait on a queue here, but is it really worth it?

Reply:
What's the upside of using a queue instead of just blocking?
I don't understand your proposal, sorry